src/Metadata/MetadataMatcher.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Metadata;
+
+use Prooph\EventStore\Exception\InvalidArgumentException;
+
+class MetadataMatcher
```

Comment:
MetadataMatcher should be an interface and this class should be named ValueMetadataMatcher instead.
In other scenarios you may want to perform more complex matches like if this value is equal to xy, another value should be abc or if metadata key exists and so on